org.idempiere.test/src/org/idempiere/test/base/InOutTest.java

```diff
@@ -500,4 +520,53 @@ public void testFreightCostRuleCustomerAccount() {
 		assertEquals(shipperAccount, inout.getShipperAccount(), "Unexpected shipper account");
 		assertEquals(MInOut.FREIGHTCHARGES_Collect, inout.getFreightCharges(), "Unexpected freight charges rule");
 	}
+	
+	/**
+	 * Test cases for Credit Check
+	 */
+	@Test
+	public void testCreditCheckInOut()
+	{
+		MBPartner bpartner = MBPartner.get(Env.getCtx(), DictionaryIDs.C_BPartner.TREE_FARM.id, getTrxName());
+		bpartner.setSOCreditStatus(MBPartner.SOCREDITSTATUS_NoCreditCheck);
+		bpartner.saveEx();
+
+		MProduct product = MProduct.get(Env.getCtx(), DictionaryIDs.M_Product.ELM.id);
+		Timestamp currentDate = Env.getContextAsDate(Env.getCtx(), "#Date");
+
+		try
+		{
+			MOrder order = createSalseOrder(bpartner, currentDate, DictionaryIDs.M_PriceList.STANDARD.id, DictionaryIDs.C_ConversionType.COMPANY.id);
+			MOrderLine orderLine = createOrderLine(order, 10, product, new BigDecimal(500), new BigDecimal(23.32));
+			completeDocument(order);
+
+			MInOut receipt = createShipment(order, currentDate);
+			BigDecimal qtyDelivered = new BigDecimal(500);
+			createInOutLine(receipt, orderLine, qtyDelivered);
+
+			ProcessInfo info = MWorkflow.runDocumentActionWorkflow(receipt, DocAction.ACTION_Prepare);
+			receipt.load(getTrxName());
+			assertTrue(info.isError(), info.getSummary());
+			assertEquals(DocAction.STATUS_Invalid, receipt.getDocStatus());
```

Comment:
why the prepare of receipt will fail here ?

Reply:
FYI.. That was bug, it fixed.